File: omegaUp/TeamSkynet-Training/pb-Figuras-Rectangulares.cpp
#include <bits/stdc++.h>
using namespace std;

inline bool is_square(int &n) {
	int root = sqrt(n);
	
	return (root * root) == n;
}

bool is_rectangle(int &n) {
	for (int i = 2; i * i < n; ++i) {
		if ((n % i) == 0) {
			return true;
			break;
		}
	}
	return false;
}

int main() {
	int n;
	bool isSquare, isRectangle;

	cin >> n;

	isSquare = is_square(n);
	isRectangle = is_rectangle(n);
	
	if (isSquare and isRectangle) {
		cout << "ambos\n";
	} else if (isSquare) {
		cout << "cuadrado\n";
	} else if (isRectangle) {
		cout << "rectangulo\n";
	} else {
		cout << "ninguno\n";
	}
	return 0;
}

File: omegaUp/TeamSkynet-Training/pb-Figuras-Rectangulares.py
from math import sqrt

def readLine():
	return raw_input()

def readInt():
	return int(readLine())

def is_square(n):
	root = int(sqrt(n))
	
	return (root * root) == n

def is_rectangle(n):
	i = 2

	while i * i < n:
		if (n % i) == 0:
			return True
			break
		i += 1

	return False
		

def main():
	n = readInt()

	if is_square(n) and is_rectangle(n):
		print('ambos')
	elif is_square(n):
		print('cuadrado')
	else:
		print('rectangulo')

if __name__ == '__main__':
	main()
